File: 104/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
    public int maxDepth(TreeNode root) {
        tree(root, 1);
        return ans;
    }
    void tree(TreeNode root, int level) {
        if(root==null) return;
        if(level>ans) ans = level;
        if(root.left!=null) tree(root.left, level+1);
        if(root.right!=null) tree(root.right, level+1);
    }
}

File: 104/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        self.ans = 0
        def helper(root, level:int):
            if root != None:
                helper(root.left, level+1)
                helper(root.right, level+1)
            if level>self.ans:
                self.ans = level
        helper(root, 0)
        return self.ans
